biz.aQute.bndlib/src/aQute/bnd/osgi/Clazz.java

```diff
@@ -784,6 +786,11 @@ public Clazz(Analyzer analyzer, String path, Resource resource) {
 		classDef = new ClassDef(classFile);
 		constantPool = classFile.constant_pool;
 		referred = new HashMap<>(constantPool.size());
+		moduleAttribute = Arrays.stream(classFile.attributes)
```

Comment:
I think this should be in ClassDef. Also, we really should not proactively get things and put the in fields. They should be obtained and returned upon demand.

Reply:
Also, this should then use the attribute methods from ElementDef.